Sources/pipeablejs/src/pipeable.ts

```diff
@@ -271,26 +280,38 @@
             return false;
         }
 
-        focus(elementHash: string) {
-            const el = this.elementRegistry.get(elementHash);
-
+        private _focus(el?: HTMLElement | Element) {
             if (el) {
-                // emit focus event.
-                const event = new FocusEvent('focus', {
-                    view: window,
-                    bubbles: true,
-                    cancelable: true,
-                });
-
-                el.dispatchEvent(event);
-                el.focus();
+                if (el instanceof HTMLElement) {
+                    (el as any).focus();
+                } else {
+                    // If the element doesn't have a focus method, we can still emit the focus event.
+                    const event = new FocusEvent('focus', {
```

Comment:
@pdobrev you were correct that `el.focus()` fires a `focus` event and a `focusin` event. So now we ensure that we only ever fire one of each. 

Not sure if this is preferred vs just firing both `focus` and `focusin` manually and once via `el.focus()`? I am actually curious why we were firing two `focus` events in the original code (e.g one via `el.dispatch(event)` and one via `el.focus()`?

AFAICT `el.focus()` is probably what we want? I found cases where `el.focus()` does not fire any events, one example is a generic `div` without `tabindex` >= 0. Do you think we should always just emit the events manually and also try to call `el.focus()` (this might be the safest thing to do)?

So firing two `focusin` events and two `focus` events with the old code. This switches it so we only ever fire one of each. However, in some cases `el.focus()` will be called but will not fire any events

Reply:
FWIW the airbnb demo works if we fire duplicate events (e.g `el.focus()`, `el.dispatch(event1)` and `el.dispatch(event2)`)

Reply:
I think generally firing more focus events is unlikely to break anything, but is weird and may eventually have some unexpected consequences if some code relies on having a listener fired exactly once.

The reason there were two (and I think this was done without extensive testing) was the reason that you mention -- el.focus() is valid only for input events, so the other events were fired manually.

Reply:
I agree firing extra focus/focusin events probably won't break anything, however I would propose that we change the behavior to the way it is in this PR (i.e we don't fire extra events) because I think it is a bit more intuitive and hopefully will save us some headaches in the future. Happy to revert the change to firing duplicate events if you think that makes more sense.

Reply:
Ah, no, this is definitely cleaner.